manifests/init.pp

```diff
@@ -42,10 +46,12 @@
   validate_re($keys_controlkey, ['^\d+$', ''])
   validate_re($keys_requestkey, ['^\d+$', ''])
   validate_array($keys_trusted)
+  validate_re($minpoll, '^([3-9]|1[0-6])$')
+  validate_re($minpoll, '^([3-9]|1[0-6])$')
```

Comment:
Rather than using `validate_re` here you should try `validate_numeric` with min/max values set ([docs](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/parser/functions/validate_numeric.rb#L3-L16)). This should also resolve the future parser issues.